### PR TITLE
More info for some throwMs

### PR DESCRIPTION
**Pull request by expede**
* Log more information when we `throwM` an error
* Remove unused helper

### Aside
This is partly to diagnose what's happening with Heroku deprovisioning. My best guess there is that we're expecting the user's login, instead of the Heroku OAuth token. More logs first though.

### CODE REVIEW

File: fission-web-server/library/Fission/Web/Server/Auth/Error.hs [hunk 1]
```diff
@@ -1,10 +1,10 @@
 module Fission.Web.Server.Auth.Error (Error (..)) where
 
-import qualified Servant.Server           as Server
+import           Servant.Server           hiding (BasicAuthResult (..))
```
Comment (walkah) [inline fission-web-server/library/Fission/Web/Server/Auth/Error.hs:3]: I assume BasicAuthResult had some conflict, but I don't see it? 🤔

File: fission-web-server/library/Fission/Web/Server/Error/Class.hs [hunk 2]
```diff
@@ -165,16 +163,16 @@ instance ToServerError JWT.Signature.Error where
 instance ToServerError JWT.Proof.Error where
   toServerError = \case
     ResolverError err     -> toServerError err
-    ScopeOutOfBounds      -> err401 { errBody = displayLazyBS ScopeOutOfBounds      }
-    PotencyEscelation     -> err401 { errBody = displayLazyBS PotencyEscelation     }
+    ScopeOutOfBounds      -> err422 { errBody = displayLazyBS ScopeOutOfBounds      }
+    PotencyEscelation     -> err422 { errBody = displayLazyBS PotencyEscelation     }
     TimeNotSubset         -> err422 { errBody = displayLazyBS TimeNotSubset         }
     MissingExpectedFact   -> err422 { errBody = displayLazyBS MissingExpectedFact   }
     InvalidSignatureChain -> err422 { errBody = displayLazyBS InvalidSignatureChain }
```
Comment (walkah) [inline fission-web-server/library/Fission/Web/Server/Error/Class.hs:170]: please tell me you chuckle every time you typed `displayLazyBS`... no?

**Review by walkah — APPROVED** [review]
🚢 👍  more info, yay!

File: fission-web-server/library/Fission/Web/Server/Auth/Error.hs (hunk 1 above)
Comment (expede) [reply fission-web-server/library/Fission/Web/Server/Auth/Error.hs:3]: `BasicAuthResult` is the type, which contains a value `NoSuchUser`, which is the conflict. I guess we could change that to `NoUserFound` or something to disambiguate, but honestly 🤷‍♀️

File: fission-web-server/library/Fission/Web/Server/Error/Class.hs (hunk 2 above)
Comment (expede) [reply fission-web-server/library/Fission/Web/Server/Error/Class.hs:170]: hahaha I totally do!

**Review by bgins — APPROVED** [review]
Awesome, more info is good! :+1:

**Pull request merged**